### PR TITLE
Rename Output to OutputFileType and deprecated Output

### PR DESCRIPTION
**Pull request by abadams**
It would be nice in future if we could promote Generator::Input and
Generator::Output to the Halide namespace. For that to happen, we need
to rename the enum Output first. This PR adds a new name for it, and
deprecates the old name. 

Not married to the name OutputFile. Other suggestions welcome.

### CODE REVIEW

File: src/Module.h [hunk 1]
```diff
@@ -43,6 +43,26 @@ enum class Output {
     stmt_html,
 };
 
+class [[deprecated("Use OutputFile instead of Output")]] Output {
```
Comment (steven-johnson) [inline src/Module.h:46]: IMHO we should still prefer to use `HALIDE_ATTRIBUTE_DEPRECATED()` here, as it allows downstream folks to disable it via `HALIDE_ALLOW_DEPRECATED`

**Review by steven-johnson — CHANGES_REQUESTED** [review]
Should update the Python versions of these enums to use the same nomenclature, see PyEnums.cpp

Re the name, OutputFile is ok with me, but also maybe OutputFileType or OutputType?

**Comment by zvookin** [comment]
I hate to ask this, and feel free to ignore if it doesn't make sense... The word "type" is heavily overloaded in a compiler project to begin with and I've come to think being careful around its use is a good idea. Specifically trying to use other words when one is not actually talking about a type in either the source or implementation language. E.g. an "output type" would be the type (buffer of ints or something) of an output.

To that end, maybe "OutputKind" is better. We're already using Kind in some places.

This is not a thing I'd object to the PR over, but figured I'd mention it in case improving our words and consistent use of them seems like a thing to do here.

**Comment by steven-johnson** [comment]
> To that end, maybe "OutputKind" is better. We're already using Kind in some places.

SGTM, +1

**Review by steven-johnson — APPROVED** [review]

**Comment by steven-johnson** [comment]
LGTM

**Comment by steven-johnson** [comment]
Win32 is apparently unrelated (see https://github.com/halide/Halide/pull/6581 for hopeful fix), LGTM to land once green

**Pull request merged**